cypress/integration/BounceRulesDetailed/bounce_rules_detailed.spec.js

```diff
@@ -0,0 +1,73 @@
+import BounceRuleDetailed from "../../Pages/BounceRulesDetailed";
+
+const testCreateRule = {
+  priority: 2,
+  bounce_action: "cypressDetailedTest",
+  response_code: 281,
+  description: "cypressDetailedTest",
+  enhanced_code: "384",
+  regex: "kjfgsdlfg",
+};
+const updatedRule = {
+  updatedDescription: "This is a new description from the Cypress Test!",
+  updatedBounceAction: "This is a new bounce action from the Cypress Test!",
+  updatedResponseCode: 999,
+  updatedEnhancedCode: "10.2.5",
+  updatedRegex: "cyress",
+  updatedCommit: "This is a new commit from a Cypress Test",
+  updatedPriority: 1,
+};
+let ruleId;
+
+describe("Bounce Rule Detailed", () => {
+  before(async () => {
+    ruleId = await BounceRuleDetailed.createTestRuleAPI(testCreateRule);
+  });
+
+  beforeEach(() => {
+    BounceRuleDetailed.open(ruleId);
+  });
+
+  after(() => {
+    BounceRuleDetailed.deleteBounceRuleAPI(ruleId).then(isCleanupSuccess => {
```

Comment:
Let's try to avoid after alls to do cleanup/teardown as the test may fail in many ways in between test cases or steps in the tests themselves and never cleanup the previous bounce rule. It's better to cleanup and teardown before the tests instead so every time it runs it's always cleaning up even if another test was possibly kicked off at the same time. We can refactor this by wrapping it in a function in the bounce rule detailed page to be less flakey and still be readable from the spec perspective.

Imagine we have a function that we can eventually chain called ```BounceRuleDetailed.teardownBounceRule(testCreateRule)```
and we'll do a get of all the bounce rules currently in the database (`cy.task("getRules", ...)`-> this will return a list of all the bounce rule objects with their rule ids. We'll run a .find on the rules to see if any of them match the testCreateRule properties (most likely just the designated test property like bounce_action of "cypressDetailedTest" - let's keep some of the properties the same when updating the changelog so regardless of whether the test was updated or not, we can clean it up safely). if we find a matching one, get that rule id and return the result of deleting that bounce rule so you can chain it; otherwise if not matching one, return true since it never existed and was already torn down.

From a higher level the before hook will look something like this
```js
BounceRuleDetailedPage.teardownBounceRule(testCreateRule).then(isTeardownSuccess => {
  if (isTeardownSuccess) {
    return cy.log("Successfully tore down test bounce rule!);
  } else {
    return cy.log("Failed to tear down test bounce rule!);
  }
}).then(() => {
  return BounceRuleDetailedPage.createTestRuleAPI(testCreateRule);
}).then((createdRuleId) => {
  if (createdRuleId) {
    cy.log(`Successfully created test bounce rule with rule ID ${ruleId}!`)'
    ruleId = createdRuleId;
   } else {
    cy.log("Failed to create test bounce rule!");
    // Still set anyways but test will most likely fail
    ruleId = createdRuleId;
   } 
})
```